### DISCUSSION TITLE
Error displaying UnderwaterAcoustics.BasicUnderwaterEnvironment

### DISCUSSION THREAD

**Discussion by pabloborys** [org-arl/UnderwaterAcoustics.jl#44] [Q&A] [answered]
Hi, I am getting this error when trying to set up the env variable. 

env = UnderwaterEnvironment()  

Error displaying UnderwaterAcoustics.BasicUnderwaterEnvironment{FlatSurface, ConstantDepth{Float64}, IsoSSP{Float64}, Float64, SurfaceLoss{Float64}, RayleighReflectionCoef{Float64, Float64, Float64}, RedGaussianNoise{Float64}}: MethodError: no method matching replace(::String, ::Pair{Regex, String}, ::Pair{Regex, String})

I am not sure if this is a local problem in my ATOM config or what?

Hope you can help, thanks a lot

**Comment by mchitre** [comment, marked as answer]
Sounds like a Julia version issue. Are you on Julia 1.6?

The error message suggests a problem with calling `replace()` with multiple patterns. From the docs:
```
replace(s::AbstractString, pat=>r, [pat2=>r2, ...]; [count::Integer])

  │ Julia 1.7
  │
  │  Support for multiple patterns requires version 1.7.
```

Since the `compat` entry marks `UnderwaterAcoustics.jl` to be compatible with Julia 1.6, this should be categorized as a bug. The CI is set up for Julia 1.6, but the display on REPL method is probably missed being covered.

I'll transfer this to an issue to fix. In the meantime, moving to Julia 1.7 should fix it for you, I think.

**Reply by mchitre** [reply]
Added issue #45 to track this.
